web_external/Datasets/UploadDatasetMetadataView.js

```diff
@@ -0,0 +1,195 @@
+            isic.showAlertDialog({
+                text: '<h4>Metadata successfully registered.</h4><br>' +
+                      'An administrator may contact you via email.',
+                escapedHtml: true,
+                callback: function () {
+                    isic.router.navigate('', {trigger: true});
+                }
+            });
+        });
+
+        this.listenTo(this.dataset, 'isic:registerMetadata:error', function (resp) {
+            isic.showAlertDialog({
+                text: '<h4>Error registering metadata</h4><br>' + resp.responseJSON.message,
+                escapedHtml: true
+            });
+            this.$('#isic-upload-dataset-metadata-submit').prop('disabled', false);
+        });
+
+        this.render();
+    },
+
+    render: function () {
+        this.$el.html(isic.templates.uploadDatasetMetadata({
+            user: girder.currentUser,
+            dataset: this.dataset
+        }));
+
+        if (!this.uploadWidget) {
+            this.initializeUploadWidget();
+        }
+        this.updateUploadWidget();
+
+        return this;
+    },
+
+    initializeUploadWidget: function () {
+        if (this.uploadWidget) {
+            this.stopListening(this.uploadWidget);
+            this.uploadWidget.destroy();
+        }
+        this.uploadWidget = new girder.views.UploadWidget({
+            parentView: this,
+            modal: false,
+            noParent: true,
+            title: false,
+            overrideStart: true
+        });
+
+        this.uploadWidget.setElement(this.$('.isic-upload-widget-container'));
+
+        this.listenTo(this.uploadWidget, 'g:filesChanged', this.filesSelected);
+        this.listenTo(this.uploadWidget, 'g:uploadStarted', this.uploadStarted);
+        this.listenTo(this.uploadWidget, 'g:uploadFinished', this.uploadFinished);
+    },
+
+    filesSelected: function (files) {
+        // TODO: could validate based on file extension
+    },
+    uploadStarted: function (files) {
+        // Prepare upload folder in user's home and start upload
+        if (this.uploadFolder) {
+            // Folder already created
+            this.startUpload(this.uploadFolder);
+        } else {
+            // Create new upload folder with unique name
+            this.uploadFolder = new girder.models.FolderModel({
+                name: 'isic_upload_' + Date.now(),
+                parentType: 'user',
+                parentId: girder.currentUser.id,
+                description: 'ISIC uploads'
+            });
+
+            this.uploadFolder.once('g:saved', function () {
+                this.startUpload(this.uploadFolder);
+            }, this).once('g:error', function () {
+                girder.events.trigger('g:alert', {
+                    icon: 'cancel',
+                    text: 'Could not create upload folder.',
+                    type: 'error',
+                    timeout: 4000
+                });
+            }, this).save();
```

Comment:
Calling `.save()` runs the request that creates the folder on the server:
https://github.com/girder/girder/blob/1.x-maintenance/clients/web/src/model.js#L32-L37